pkg/server/http/http.go

```diff
@@ -68,6 +71,29 @@ func New(logger log.Logger, reg *prometheus.Registry, comp component.Component,
 	}
 }
 
+// RegisterDownscale registers HTTP handler compatible with Grafana rollout operator.
+// See https://github.com/databricks/rollout-operator?tab=readme-ov-file#delayed-scaledown.
+func RegisterDownscale[K comparable, V any](s *Server, m map[K]V, mtx *sync.RWMutex, t *int64) {
+	s.mux.Handle("/-/downscale", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		mtx.RLock()
+		n := len(m)
```

Comment:
Don't pass in a map, if only the size of map is needed. Pass in the size of the map.

Reply:
the size of the map changes dynamically during handler runtime and [Golang passes map by reference ](https://stackoverflow.com/questions/40680981/are-maps-passed-by-value-or-by-reference-in-go)

Reply:
refactored code